# Практическая работа №4: Элементы корреляционного анализа. Проверка статистической гипотезы о равенстве коэффициента корреляции нулю

Выполнил студент гр. 1384 Степаненко Денис Владиславович.

## Цель работы

Освоение основных понятий, связанных с корреляционной зависимостью между случайными величинами, статистическими гипотезами и проверкой их «справедливости».

## Основные теоретические положения

Если случайны величины независимы, то $f(x,y)=f_1(x)\cdot f_2(y)$ иначе между ними может быть:

1) функциональная зависимость:
$y = g(x)$

2) статистическая зависимость:
$\phi (x|y) = f(x,y)/f_2(y)$

$\phi (y|x) = f(x,y)/f_1(x)$

Корелляционная зависимость - статистичеакая зависимость.

**Коэффициент корелляции**
$$r_{xy} = \frac{\mu_{xy}}{\sigma_x \sigma_y},$$
для него справедливо: $|r_{xy}|\le 1$'

Справедливы утверждения:

X и Y кореллированы -> СВ X и Y зависимы

X и Y независимы -> СВ X и Y некореллированы




## Постановка задачи

Из заданной генеральной совокупности сформировать выборку по второму признаку. Провести статистическую обработку второй выборки в объёме практических работ №1 и №2, с целью определения точечных статистических оценок параметров распределения исследуемого признака (математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса и коэффициента вариации). Для системы двух случайных величин $X$ (первый признак) и $Y$ (второй признак) сформировать двумерную выборку и найти статистическую оценку коэффициента корреляции, построить доверительный интервал для коэффициента корреляции и осуществить проверку статистической гипотезы о равенстве коэффициента корреляции нулю. Полученные результаты содержательно проинтерпретировать.

## Выполнение работы

## Пункт 1.
Провести статистическую обработку второй выборки в объёме практических работ №1 и №2, с целью определения точечных статистических оценок параметров распределения исследуемого признака (математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации). Оформить результаты в виде таблицы, сделать выводы.

Данные рассчеты уже были проведены и проанализированы в соответсвующих лабораторных работах. Отобразим результаты в виде таблицы.

|$\bar X$ |$\mathbb{D_в}$| $\sigma_в$  |$\bar A_s$ | $\bar E$ | $M_o^*$ |$M_e^*$ | $V^*$ |
|---|---|---|---|---|---|---|---|
|  2.6584 |  0.485 | 0.6964  |  -1.0697 |  0.3039 |  3.1355 | 2.8697  |  26% |


### Пункт 2.
Построить двумерный интервальный вариационный ряд, оформить в виде таблицы.

|  |$[42, 58.9485)$| $[58.9485,  75.897 )$| $[75.897,   92.8455)$ | $[92.8455, 109.7939)$ | $[109.7939, 126.7424)$ | $[126.7424, 143.6909)$ |$[143.6909, 160.6394]$ |
|---|---|---|---|---|---|---|---|
| $[1    , 1.4286)$  | 5 | 0 | 1 | 1 | 9 | 1 | 3 |
| $[1.4286, 1.8571)$ | 3 | 0 | 0 | 0 | 14 | 0 | 2 |
| $[1.8571, 2.2857)$ | 4 | 1 | 0 | 1 | 12 | 0 | 0 |
| $[2.2857, 2.7143)$ | 2 | 0 | 2 | 4 | 7 | 0 | 1 |
| $[2.7143, 3.1429)$ | 0 | 1 | 0 | 0 | 17 | 0 | 0 |
| $[3.1429, 3.5714]$ | 0 | 0 | 0 | 0 | 1 | 0 | 0 |
| $[3.5714, 4    ]$  | 0 | 0 | 0 | 0 | 2 | 0 | 0 |

In [2]:
import numpy as np
import pandas as pd

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
df = pd.read_csv("/content/drive/MyDrive/obesity_data.csv")
data = np.array(df)

In [35]:
k = round(1 + int(np.log2(len(df))))
bins1 = pd.cut(df["Weight"], bins=k, right=False, precision=6)
bins2 = pd.cut(df["NCP"], bins=k, right=False, precision=6)

contingency_table_print = pd.crosstab(bins2, bins1,  margins=True, margins_name="Total")
contingency_table_use = pd.crosstab(bins2, bins1)
contingency_table_print.rename(index={"Total": "$n_x$"}, columns={"Total": "$n_y$"}, inplace=True)
markdown_table = contingency_table_use.to_markdown()
print(markdown_table)

| NCP                  |   [42.0, 58.948486) |   [58.948486, 75.896973) |   [75.896973, 92.845459) |   [92.845459, 109.793946) |   [109.793946, 126.742432) |   [126.742432, 143.690919) |   [143.690919, 160.758044) |
|:---------------------|--------------------:|-------------------------:|-------------------------:|--------------------------:|---------------------------:|---------------------------:|---------------------------:|
| [1.0, 1.428571)      |                   5 |                        3 |                        4 |                         2 |                          0 |                          0 |                          0 |
| [1.428571, 1.857143) |                   0 |                        0 |                        1 |                         0 |                          1 |                          0 |                          0 |
| [1.857143, 2.285714) |                   1 |                        0 |                        0 |                         2 |        

### Пункт 3.
По полученному двумерному интервальному вариационному ряду построить корреляционную таблицу, сделать выводы.

In [36]:
markdown_table = contingency_table_print.to_markdown()
print(markdown_table)

| NCP                  |   [42.0, 58.948486) |   [58.948486, 75.896973) |   [75.896973, 92.845459) |   [92.845459, 109.793946) |   [109.793946, 126.742432) |   [126.742432, 143.690919) |   [143.690919, 160.758044) |   $n_y$ |
|:---------------------|--------------------:|-------------------------:|-------------------------:|--------------------------:|---------------------------:|---------------------------:|---------------------------:|--------:|
| [1.0, 1.428571)      |                   5 |                        3 |                        4 |                         2 |                          0 |                          0 |                          0 |      14 |
| [1.428571, 1.857143) |                   0 |                        0 |                        1 |                         0 |                          1 |                          0 |                          0 |       2 |
| [1.857143, 2.285714) |                   1 |                        0 |                       

Таблица 2

| NCP                  |   [42.0, 58.948486) |   [58.948486, 75.896973) |   [75.896973, 92.845459) |   [92.845459, 109.793946) |   [109.793946, 126.742432) |   [126.742432, 143.690919) |   [143.690919, 160.758044) |   $n_y$ |
|:---------------------|--------------------:|-------------------------:|-------------------------:|--------------------------:|---------------------------:|---------------------------:|---------------------------:|--------:|
| [1.0, 1.428571)      |                   5 |                        3 |                        4 |                         2 |                          0 |                          0 |                          0 |      14 |
| [1.428571, 1.857143) |                   0 |                        0 |                        1 |                         0 |                          1 |                          0 |                          0 |       2 |
| [1.857143, 2.285714) |                   1 |                        0 |                        0 |                         2 |                          0 |                          0 |                          0 |       3 |
| [2.285714, 2.714286) |                   1 |                        0 |                        1 |                         4 |                          0 |                          0 |                          0 |       6 |
| [2.714286, 3.142857) |                   9 |                       14 |                       12 |                         7 |                         17 |                          1 |                          2 |      62 |
| [3.142857, 3.571429) |                   1 |                        1 |                        0 |                         0 |                          0 |                          0 |                          0 |       2 |
| [3.571429, 4.003)    |                   3 |                        2 |                        0 |                         1 |                          0 |                          0 |                          0 |       6 |
| $n_x$                |                  20 |                       20 |                       18 |                        16 |                         18 |                          1 |                          2 |      95 |

По таблице видно, что суммы частоты по всем строкам и стобцам совпадают - вычисления верны. Также можно заметить, что нет выраженной зависимости между столбцами и строками - большинство людей ест около 3 раз в день, хотя и имет разный вес.

### Пункт 4.
Исходя из результатов корреляционной таблицы вычислить значение выборочного коэффициента корреляции двумя способами: с помощью стандартной формулы и с помощью условных вариант. Убедиться, что результаты совпадают. Сделать выводы.

$\bar r_{xy} = \frac{\sum_{i=1}^{K_y} \sum_{j=1}^{K_x} n_{ij}y_ix_j - N\bar x_в \bar y_в}{NS_xS_y}$

In [38]:
intervals1 = bins1.cat.categories
intervals2 = bins2.cat.categories

n_x = np.array(contingency_table_use.sum(axis=0))
n_y = np.array(contingency_table_use.sum(axis=1))

x = np.array([(interval.left + interval.right) / 2 for interval in intervals1])
y = np.array([(interval.left + interval.right) / 2 for interval in intervals2])

contingency_table = np.array(contingency_table_use)
N = np.sum(contingency_table)

x_mean = np.sum(n_x*x)/N
D_x = 1/N*(np.sum(n_x*(x - x_mean)**2))
S_x = np.sqrt((N-1)/N*D_x)

y_mean = np.sum(n_y*y)/N
D_y = 1/N*(np.sum(n_y*(y - y_mean)**2))
S_y = np.sqrt((N-1)/N*D_y)

sum = 0
for i in range(len(contingency_table[0])):
  for j in range(len(contingency_table[1])):
    sum += contingency_table[i,j]*x[i]*y[j]

cov = (sum - N*x_mean*y_mean)/N

r_xy = cov/(S_x*S_y)
print(r_xy)


0.10402525999171791


In [39]:
C_x = x[np.argmax(n_x)]
C_y = y[np.argmax(n_y)]

h_x = (df["Weight"].max() - df["Weight"].min())/k
h_y = (df["NCP"].max() - df["NCP"].min())/k

u = (x-C_x)/h_x
v = (y-C_y)/h_y

u_mean = np.sum(u*n_x)/N
v_mean = np.sum(v*n_y)/N

u2_mean = np.sum(u**2*n_x**2)/N
v2_mean = np.sum(v**2*n_y**2)/N

sigma_u = np.sqrt(u2_mean - u_mean**2)
sigma_v = np.sqrt(v2_mean - v_mean**2)

sum = 0
for i in range(len(contingency_table[0])):
  for j in range(len(contingency_table[1])):
    sum += contingency_table[i,j]*u[i]*v[j]

cov = (sum - N*u_mean*v_mean)/N

r_xy = cov/(sigma_u*sigma_v)
print(r_xy)

0.10402525999171791


Значение выборочного коэффициента корреляции вычислено двумя способами:
С помощью стандартной формулы и с помощью условных вариант. Оба метода дают одинаковый результат: $r_{xy}\approx 0.104$. Низкое значение коэффициента корреляции указывает на слабую линейную зависимость между признаками первой и второй выборки.

### Пункт 5.
Построить доверительный интервал для коэффициента корреляции при уровне значимости
γ
∈
{
0.95
,
0.99
}
, сделать выводы.

In [40]:
import scipy.stats as stats

def confidence_interval_r(r_xy, N, gamma):
    z = 0.5 * np.log((1 + r_xy)/(1 - r_xy))
    alpha = 1 - gamma
    _lambda = stats.norm.ppf(1 - alpha / 2)

    left  = z - _lambda / np.sqrt(N - 3)
    right = z + _lambda / np.sqrt(N - 3)

    left  = (np.exp(2 * left) - 1) / (np.exp(2 * left) + 1)
    right = (np.exp(2 * right) - 1) / (np.exp(2 * right) + 1)

    return left, right

left_95, right_95 = confidence_interval_r(r_xy, N, 0.95)
left_99, right_99 = confidence_interval_r(r_xy, N, 0.99)

print(f"Доверитиельный интервал r: ({left_95:.4f}; {right_95:.4f})")
print(f"Доверитиельный интервал r: ({left_99:.4f}; {right_99:.4f})")

Доверитиельный интервал r: (-0.0996; 0.2993)
Доверитиельный интервал r: (-0.1627; 0.3566)


Интервалы содержат ноль, что подтверждает гипотезу о возможном отсутствии значимой корреляции между признаками.Широкий диапазон интервалов объясняется небольшим объемом выборки N=95. Доверительный интеравал при уровне значимости $\gamma = 0.99$ ожидаемо оказался больше.

### Пункт 6.
Осуществить проверку статистической гипотезы о равенстве коэффициента корреляции нулю при заданном уровне значимости
α
=
0.05
, сделать выводы.

In [41]:
# проверка гипотезы
alpha = 0.05
t_sample = r_xy * np.sqrt(N-2) / np.sqrt(1 - r_xy**2)
t_crit = stats.t.ppf(1-alpha, N-2)

print(f't выборочное  = {t_sample:.4f}')
print(f't критическое = {t_crit:.4f}')
if t_sample <= t_crit:
    print('H0 не отвергается')
else:
    print('Отвергаем гипотезу H0')

t выборочное  = 1.0087
t критическое = 1.6614
H0 не отвергается


Поскольку $t_{выб} <t_{крит}$ , нет оснований отвергать нулевую гипотезу.Таким образом, можно сделать вывод, что коэффициент корреляции статистически не отличается от нуля.

## Выводы

В ходе выполнения лабораторной работы была проведена статистическая обработка второй выборки с использованием программных средств, которые были разработаны в рамках предыдущих лабораторных работ (№1–2). Это позволило собрать необходимые данные для исследования корреляционной зависимости между изучаемыми признаками.

Для анализа взаимосвязи между величинами была сформирована двумерная выборка и построена корреляционная таблица. На основе полученных данных был рассчитан коэффициент корреляции Пирсона, который составил $r_в\approx 0.104$. Этот результат указывает на очень слабую линейную связь между исследуемыми параметрами.

Чтобы оценить значимость этого значения, был построен доверительный интервал для коэффициента корреляции при уровне доверия 95% и 99%. Интервалы оказался равным $(-0.1914; 0.2116)$  и  $(-0.2524; 0.2721)$ соответственно. Поскольку ноль попадает в этот диапазон, можно сделать вывод, что корреляционная зависимость между величинами не является значимой.

Дополнительно была проверена гипотеза о равенстве коэффициента корреляции нулю $H_0: r_xy = 0$ при уровне значимости $\alpha=0.05$. Для этого были вычислены выборочное значение статистики критерия $t_выб=1.0087$ и критическое значение $t_кр=1.6614$. Так как выборочное значение оказалось меньше критического, нет оснований отвергать нулевую гипотезу. Это подтверждает, что коэффициент корреляции статистически не отличается от нуля, и значимая линейная зависимость между исследуемыми параметрами отсутствует.

Таким образом, проведенное исследование показало, что между признаками X (вес) и Y (NCP) отсутствует значимая линейная корреляционная зависимость. Полученные результаты согласуются с гипотезой о независимости данных величин.